cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -136,7 +137,15 @@ public ResourceGroups getResourceGroups() {
     }
 
     public boolean resourceGroupExists(String name) {
-        return getResourceGroups().contain(name);
+        try {
+            return getResourceGroups().contain(name);
+        } catch (CloudException e) {
+            if (e.getMessage().contains("Status code 403")) {
```

Comment:
This does not look that nice but I assume you've found no better way to distinguish the error type, right?